File: legacy/_1.7.10/src/main/java/cn/enaium/noexpensive/callback/impl/EnchantmentCanCombineCallbackImpl.java
package cn.enaium.noexpensive.callback.impl;

import cn.enaium.noexpensive.Config;
import cn.enaium.noexpensive.callback.EnchantmentCanCombineCallback;
import net.minecraft.enchantment.Enchantment;

import java.util.List;
import java.util.Map;

/**
 * @author Enaium
 */
public class EnchantmentCanCombineCallbackImpl implements EnchantmentCanCombineCallback {
    @Override
    public boolean interact(Enchantment enchantment1, Enchantment enchantment2) {
        final Map<Integer, List<Integer>> compatibility = Config.getModel().compatibility;

        if (compatibility.containsKey(enchantment1.id) && compatibility.get(enchantment1.id).contains(enchantment2.id)) {
            return true;
        } else if (compatibility.containsKey(enchantment2.id) && compatibility.get(enchantment2.id).contains(enchantment1.id)) {
            return true;
        }
        return enchantment1.differs(enchantment2);
    }
}

File: legacy/_1.7.10/src/main/kotlin/cn/enaium/noexpensive/callback/impl/EnchantmentCanCombineCallbackImpl.kt
package cn.enaium.noexpensive.callback.impl

import cn.enaium.noexpensive.Config
import cn.enaium.noexpensive.callback.EnchantmentCanCombineCallback
import net.minecraft.enchantment.Enchantment


/**
 * @author Enaium
 */
class EnchantmentCanCombineCallbackImpl : EnchantmentCanCombineCallback {
    override fun interact(enchantment1: Enchantment, enchantment2: Enchantment): Boolean {
        val compatibility: Map<Int, List<Int>> = Config.model.compatibility
        if (compatibility.containsKey(enchantment1.id) && compatibility[enchantment1.id]!!.contains(enchantment2.id)) {
            return true
        } else if (compatibility.containsKey(enchantment2.id) && compatibility[enchantment2.id]!!.contains(enchantment1.id)) {
            return true
        }
        return enchantment1.differs(enchantment2)
    }
}
